tests/test_pytorch/test_fsdp.py

```diff
@@ -0,0 +1,791 @@
+# Copyright (c) 2023 Habana Labs, Ltd. an Intel Company
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from functools import partial
+from pathlib import Path
+from typing import Optional, Tuple
+from unittest import mock
+from unittest.mock import ANY, MagicMock, Mock
+
+import pytest
+import torch
+import torch.nn as nn
+from lightning_utilities import module_available
+from torch.distributed.fsdp.fully_sharded_data_parallel import CPUOffload, FullyShardedDataParallel, MixedPrecision
+from torch.distributed.fsdp.wrap import size_based_auto_wrap_policy, wrap
+from torchmetrics import Accuracy
+
+if module_available("lightning"):
+    from lightning.fabric.utilities.imports import (
+        _TORCH_GREATER_EQUAL_2_0,
+        _TORCH_GREATER_EQUAL_2_1,
+    )
+    from lightning.pytorch import Trainer, seed_everything
+    from lightning.pytorch.demos import LightningTransformer
+    from lightning.pytorch.demos.boring_classes import BoringDataModule, BoringModel
+elif module_available("pytorch_lightning"):
+    from lightning_fabric.utilities.imports import (
+        _TORCH_GREATER_EQUAL_2_0,
```

Comment:
do we need 2.0 ?

Reply:
We don't need check for 2.0. I have removed it from our tests